arch/arm/include/stm32h7/chip.h

```diff
@@ -53,16 +53,40 @@
  *
  *   PART        PACKAGE
  *   ----------- ----------------
+ *   STM32H7x3Ax UFBGA169
```

Comment:
@hartmannathan 

> Where is that addressed?

Grep for STM32H7_NGPIO then look at the F7 and follow the pattern.

For the package with different pin counts we do not want to make something that is not whole appear to be.  Or said another way: A problem found at compile time is much better then one found in the debugger.

Reply:
@davids5 How would you define STM32H7_NGPIO if there are gaps in the ports? For example, if ports A, B, C, D, E, H are exposed to pins, and ports F and G are not exposed to pins, is STM32H7_NGPIO supposed to be defined to 6 or 8?

Reply:
@hartmannathan - what part number are you referring to? Is H just the OSC?

Reply:
@davids5 - any of the 100-pin "V" parts have this condition, e.g., STM32H743VITx. Yes, H is only PH0-OSC_IN, PH1-OSC_OUT. But in addition to oscillator, these pins can be GPIO.

Reply:
@davids5 - The latest commit I pushed to this PR fixes the TODO IO_CONFIG items in the STM32H7 Kconfig and defines STM32H7_NGPIO according to those IO_CONFIGs. Please give me your opinion on this.

Also, as I mentioned before, the 100-pin IO_CONFIG_V parts have a gap in the port letters, it has GPIOA-GPIOH, but it's missing GPIOF-GPIOG. Should this be handled, and what would you recommend?

Thanks

Reply:
@hartmannathan - 


![image](https://user-images.githubusercontent.com/1945821/79220829-9d794f80-7e09-11ea-82cd-6ecc48113b75.png)

I would think the RCC sets up the OSC is enabled and From the above, If I have the correct datasheet I do not see any AF0-14 for PH0 and 1,

Reply:
Yes, in all the STM32H7x3xx, PH0 and PH1 have only one alternate function: AF15, EVENTOUT. And, these pins can be GPIOs if not used for the OSC.

So, which of the following would you prefer?

- **STM32H7_NGPIO** defined to 6 (GPIOA thru GPIOF) and ignore GPIOH, or
- **STM32H7_NGPIO** defined to 8 (GPIOA thru GPIOH), and add a **CONFIG_STM32H7_HAVE_GPIOG** to Kconfig.

Reply:
I think the CONFIG_STM32H7_HAVE_GPIOx solution is good way to go, but to catch it at compile time, the CONFIG_STM32H7_HAVE_GPIOx will have to wrapped in the #if defined(CONFIG_STM32H7_HAVE_GPIOx). From the run time side, the table could be set to null and an error returned.

Reply:
Commit 1dd150b ("stm32h7: Handle missing GPIO ports F and G on 100-pin parts") adds this handling. Could you please review the changes?